### PR TITLE
Add valid move queen

### PR DESCRIPTION
**Pull request by zenkbaries**
I think I got the code down and the test passed...Initially

but i need to create more testing for different scenario and not really how to "FactoryGirl" that scenario.

Any help would be appreciated!

### CODE REVIEW

File: spec/models/queen_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    end
+
+    it 'return false for not valid move' do
+      queen = FactoryGirl.create(:queen)
+      expect(queen.valid_move?(2, 6)).to eq(false)
+    end
+
+    it 'return false because destination has our pieces' do
+      queen = FactoryGirl.create(:queen)
+      expect(queen.valid_move?(2.6)).to eq(false)
```
Comment (mscoutermarsh) [inline spec/models/queen_spec.rb]: typo :)

`expect(queen.valid_move?(2,6)).to eq(false)`

**Comment by zenkbaries** [comment]
@mscoutermarsh  So this is ready for merge.... do I go ahead merge. Or you want to review it first?

File: app/models/scratch [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+
+[1, 0],   = 0
+[-1, 0],  =
+[0, -1],  =
+[0, 1],   =
+[1, -1],  =
+[1, 1],   =
+[-1, -1], =
+[-1, 1]   =
```
Comment (mscoutermarsh) [inline app/models/scratch]: can you remove this file from git :)

**Comment by mscoutermarsh** [comment]
Looks great. Just that 1 extra file that's not needed. You can merge whenever you're ready

**Pull request merged**